ontopy/ontology.py

```diff
@@ -1270,6 +1273,60 @@ def addancestors(entity, counter, subject):
             return ancestors.difference(classes)
         return ancestors
 
+    def get_descendants(
+        self,
+        classes: "Union[List, ThingClass]",
+        common: bool = False,
+        generations: int = -1,
+        all_descendants: bool = False,
+    ):
+        """Return descendants/subclasses of all classes in `classes`.
+        Arguments:
+        - classes: to be provided as list.
+        - common: only return common descendants.
+        - generations: Include this number of descendant levels.
+        - all_descendants: Include all descendants.
+        Returns:
+            A list of descendants including required generation.
+            If 'common'=True, the common descendants are returned
+            within the specified number of generations.
+            If 'generations' is not given and 'common' is True all
+            descendants will be checked.
+            'generations' defaults to 1 if 'common' not True.
+        """
+        if (common) & (generations == -1):
+            all_descendants = True
+        elif generations == -1:
+            generations = 1
```

Comment:
Is there a particular reason for this logic? I would suggest to remove it, to decouple the arguments from each other.

Reply:
Because if we ask for only common descendants it is more likely that we are interested in all generations. But if we change default to all generations this goes out anyhow. I'll remove it.